homeassistant/components/starlink/sensor.py

```diff
@@ -0,0 +1,205 @@
+"""Contains sensors exposed by the Starlink integration."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorStateClass
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DATA_RATE_BITS_PER_SECOND,
+    DATA_RATE_MEGABITS_PER_SECOND,
+    DEGREE,
+    TIME_MILLISECONDS,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONFIG_LINK_SPEED_FORMAT, DOMAIN
+from .coordinator import StarlinkUpdateCoordinator
+from .entity import StarlinkSensorEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up all sensors for this entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    config_name = entry.title
+    config_speed_format = entry.data[CONFIG_LINK_SPEED_FORMAT]
+
+    async_add_entities(
+        [
+            PingSensor(coordinator=coordinator, config_name=config_name),
+            AzimuthSensor(coordinator=coordinator, config_name=config_name),
+            ElevationSensor(coordinator=coordinator, config_name=config_name),
+            UplinkThroughputSensor(
+                coordinator=coordinator,
+                config_name=config_name,
+                data_unit=config_speed_format,
+            ),
+            DownlinkThroughputSensor(
+                coordinator=coordinator,
+                config_name=config_name,
+                data_unit=config_speed_format,
+            ),
+            LastBootTime(coordinator=coordinator, config_name=config_name),
+        ]
+    )
+
+
+class PingSensor(StarlinkSensorEntity):
+    """A sensor exposing Dishy's latency in milliseconds."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_native_unit_of_measurement = TIME_MILLISECONDS
+    _attr_icon = "mdi:speedometer"
+    _attr_unique_id = "ping"
+
+    def __init__(
+        self, coordinator: StarlinkUpdateCoordinator, config_name: str
+    ) -> None:
+        """Initialize the sensor and set the update coordinator."""
+        self._attr_name = config_name + " Ping"
+        super().__init__(coordinator)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
```

Comment:
Since you use `CoordinatorEntity` you don't need to override `_handle_coordinator_update()`. You can override `native_value()` - it will be called automatically when coordinator updates.